code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LDAPIdentityProvider.java

```diff
@@ -239,14 +239,23 @@ protected boolean isType(Attributes search, String type) {
         }
     }
 
-    protected NamingEnumeration<SearchResult> searchLdap(String query) {
+    protected NamingEnumeration<SearchResult> searchLdap(String query, String scope) {
         SearchControls controls = new SearchControls();
         LdapContext context = null;
         controls.setSearchScope(SUBTREE_SCOPE);
         NamingEnumeration<SearchResult> results;
         try {
             context = getServiceContext();
-            results = context.search(getConstantsConfig().getDomain(), query, controls);
+            if (getConstantsConfig().getGroupScope().equalsIgnoreCase(scope)) {
+                String groupSearchDomain = getConstantsConfig().getGroupSearchDomain();
+                if (StringUtils.isNotBlank(groupSearchDomain)) {
+                    results = context.search(getConstantsConfig().getGroupSearchDomain(), query, controls);
+                } else {
+                    results = context.search(getConstantsConfig().getDomain(), query, controls);
+                }
+            } else {
+                results = context.search(getConstantsConfig().getDomain(), query, controls);
+            }
```

Comment:
@cjellick yes my bad, changed it.
There were no separate AD validation tests, but there are ldap tests. We decided to add this to OpenLDAP later. I can go through existing ldap tests and add test for this